### ISSUE TITLE
Specify GKE log format using pod labels [feature request]

### ISSUE THREAD

**Issue by tmshn** [GoogleCloudPlatform/fluent-plugin-google-cloud#284]
# TL; DR

I want to specify format (namely, regex might be enough) for logs from GKE containers.

# Background

Currently GKE container logs can be in two formats: plain text or JSON. In plain text format, severity and time fields are determined irrelevant to the logs themselves. If you want control those fields, you have to emit logs in JSON format.

However, sometimes it's hard to control log format (imagine Nginx error logs or Redis logs).

I know I can achieve this by deploying sidecar container to reform records, but it's too bothering.

# How is it possible

Updating fluentd configMap every time new pods are deployed? Maybe not, it's not the clean way.

So I propose to parse log entry right after parsing it as JSON.

https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/2ea4c6b399509a91d89a2f51fd2979b84647fe2a/lib/fluent/plugin/out_google_cloud.rb#L632

```ruby
record_json = parse_json_or_nil(record[field]) || parse_regex_or_nil(record[field], regex)
```

And it's handy if we can specify the regex using pod labels:

```yaml
metadata:
  labels:
    name: front
    google-fluentd.nginx.stderr.format: ^(?<time>.+?) \[(?<severity>.*?)\] (?<pid>\d+)#(?<tid>\d+): (?<message>.*)$
spec:
  containers:
  - name: nginx
```

I believe this is possible because I can see pod labels in logging (although I couldn't find where this `userLabels` are set):

<img width="390" alt="pod-label" src="https://user-images.githubusercontent.com/3760893/51423244-67e2b800-1c00-11e9-968c-979869daff68.png">
<img width="431" alt="log-metadata" src="https://user-images.githubusercontent.com/3760893/51423247-70d38980-1c00-11e9-8d5c-48aa97f17891.png">

**Comment by jkohen** [comment]
Hi, thanks for the a feature request. This seems to be a feature specific to Google Kubernetes Engine. We don't control how the plugin is deployed, please file a feature request on the Google Kubernetes Engine console, or through your account manager if you have one.
